core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1118,7 +1118,7 @@ private[spark] object Utils extends Logging {
     // This means some invalid addresses are treated as v6 addresses, but since they are
     // not valid hostnames it doesn't matter.
     // See https://www.rfc-editor.org/rfc/rfc1123#page-13 for context around valid hostnames.
-    val addressRe = """^\[{0,1}([0-9:]+?:[0-9]+)\]{0,1}$""".r
+    val addressRe = """^\[{0,1}([0-9:]+?:[0-9]*)\]{0,1}$""".r
```

Comment:
Is there any reason why not use just this tiny regexp:
```
"""^\[{0,1}([0-9:]+)\]{0,1}$"""
```

Reply:
So that won't guarantee that there is at least one : in it, which we want since ":" is not a valid part of a hostname but is required for a v6 address.

Reply:
Absolutely true!